sdk/protocol-plugins/src/plugins/morphoblue/implementation/MorphoProtocolPlugin.ts

```diff
@@ -0,0 +1,421 @@
+      type: PoolType.Lending,
+      id: morphoLendingPoolId,
+      collateral: collateralInfo,
+      debt: debtInfo,
+    })
+  }
+
+  /** POSITIONS */
+
+  /** @see BaseProtocolPlugin.getPosition */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async getPosition(positionId: IMorphoPositionIdData): Promise<Position> {
+    this._validatePositionId(positionId)
+
+    throw new Error(`Not implemented ${positionId}`)
+  }
+
+  /** IMPORT POSITIONS */
+
+  /** @see BaseProtocolPlugin.getImportPositionTransaction */
+  async getImportPositionTransaction(params: {
+    user: IUser
+    externalPosition: IExternalPosition
+    positionsManager: IPositionsManager
+  }): Promise<Maybe<TransactionInfo>> {
+    throw new Error(`Not implemented ${params}`)
+  }
+
+  /** PRIVATE */
+
+  /**
+   * @name _getCollateralInfo
+   * @description Get the collateral info for the given morpho lending pool ID
+   * @param params The parameters
+   * @returns The collateral info
+   */
+  private async _getCollateralInfo(params: {
+    morphoLendingPool: IMorphoLendingPool
+    marketInfo: MorphoMarketInfo
+    marketCollateralPriceInDebt: IPrice
+  }): Promise<CollateralInfo> {
+    const { morphoLendingPool, marketInfo, marketCollateralPriceInDebt } = params
+
+    const collateralToken = morphoLendingPool.collateralToken
+    const liquidationPenalty = this._getLiquidationPenalty(morphoLendingPool)
+    const liquidationThreshold = this._getLiquidationThreshold(
+      marketInfo,
+      marketCollateralPriceInDebt,
+    )
+
+    const collateralPriceUSD = await this.ctx.oracleManager.getSpotPrice({
+      baseToken: collateralToken,
+    })
+
+    return CollateralInfo.createFrom({
+      token: collateralToken,
+      price: marketCollateralPriceInDebt,
+      priceUSD: collateralPriceUSD.price,
+      liquidationThreshold: liquidationThreshold,
+      tokensLocked: marketInfo.totalSupplyAssets,
+      maxSupply: TokenAmount.createFrom({
+        token: collateralToken,
+        amount: Number.MAX_SAFE_INTEGER.toString(),
+      }),
+      liquidationPenalty: liquidationPenalty,
+    })
+  }
+
+  /**
+   * @name _getDebtInfo
+   * @description Get the debt info for the given morpho lending pool ID
+   * @param morphoLendingPoolId The lending pool ID
+   * @param marketInfo The market info
+   * @param marketCollateralPriceInDebt The market collateral price in debt
+   * @returns The debt info
+   */
+  private async _getDebtInfo(params: {
+    morphoLendingPool: IMorphoLendingPool
+    marketInfo: MorphoMarketInfo
+    marketCollateralPriceInDebt: IPrice
+  }) {
+    const { morphoLendingPool, marketInfo, marketCollateralPriceInDebt } = params
+
+    const debtToken = morphoLendingPool.debtToken
+    const priceUSD = await this.ctx.oracleManager.getSpotPrice({
+      baseToken: debtToken,
+    })
+
+    const debtCeiling = marketCollateralPriceInDebt
+      .multiply(marketInfo.totalSupplyAssets)
+      .multiply(morphoLendingPool.lltv)
```

Comment:
I think we might need to subtract current total pool debt, but I'm not sure, it depends how we use it.